### PR TITLE
LPD-15350 Autogenerate headless ts clients

### PR DESCRIPTION
**Pull request by 4lejandrito**
Follow up of https://github.com/liferay-headless/liferay-portal/pull/1835.

This is how it looks like now:

```ts
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com/
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

// @ts-ignore

import {expect, mergeTests} from '@playwright/test';
import * as path from 'path';

import {documentLibraryPagesTest} from '../../fixtures/documentLibraryPages.fixtures';
import {exportImportPagesTest} from '../../fixtures/exportImportPages.fixtures';
import {loginTest} from '../../fixtures/loginTest';
import {HeadlessDeliveryClient} from '../../../../apps/headless/headless-delivery/headless-delivery-client-js/src/main/resources/META-INF/resources/node';

export const test = mergeTests(
	documentLibraryPagesTest,
	exportImportPagesTest,
	loginTest
);

test('can import a folder with document type restrictions and workflow', async ({
	documentLibraryEditFolderPage,
	documentLibraryPage,
	exportImportFramePage,
	authenticate,
}) => {
	await documentLibraryPage.goto();
	await documentLibraryPage.openOptionsMenu();
	await documentLibraryPage.exportImportOptionsMenuItem.click();
	await exportImportFramePage.importLARFile(
		path.join(__dirname, 'dependencies', 'folder.portlet.lar')
	);
	await exportImportFramePage.close();
	await documentLibraryPage.editEntry('LPS-205933');

	expect(
		await documentLibraryEditFolderPage.getSelectedWorkflowDefinition()
	).toBe('Single Approver@1');

	await authenticate(
		HeadlessDeliveryClient
	).documentFolder.deleteSiteDocumentsFolderByExternalReferenceCode({
		siteId: 'Guest' as any,
		externalReferenceCode: 'LPS-205933',
	});
});
```
**Todo:**

1. [ ] Apply to all the tests (I just did export-import and layout).
2. [ ] I found that, for some reason, the [ObjectFolder schema is empty](https://github.com/4lejandrito/liferay-portal/blob/4ea986c771a564e1864ffe18a9c57a2b899b90d1/modules/apps/object/object-admin-rest-client-js/src/main/resources/META-INF/resources/node/models/ObjectFolder.ts#L7-L6) for the object-admin-rest module. Which makes it impossible to interact with that API. It was fine with the original generated code.
3. [ ] Fix SF.

### CODE REVIEW

File: modules/test/playwright/tests/export-import-web/import.spec.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ test('can import a folder with document type restrictions and workflow', async (
 	await authenticate(
 		HeadlessDeliveryClient
 	).documentFolder.deleteSiteDocumentsFolderByExternalReferenceCode({
+		// TODO: This can also be a string but the type only supports number
```
Comment (4lejandrito) [inline modules/test/playwright/tests/export-import-web/import.spec.ts:44]: @drewbrokke ignore this for now. It is not true. I will remove it next week.

**Comment by drewbrokke** [comment]
> - [ ] I found that, for some reason, the ObjectFolder schema is empty for the object-admin-rest module. Which makes it impossible to interact with that API. It was fine with the original generated code.
This issue should be addressed by the validation introduced in https://github.com/liferay-headless/liferay-portal/pull/1840.

**Comment by 4lejandrito** [comment]
Rebased and fixed conflicts.

**Comment by 4lejandrito** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/146840

**Pull request closed without merging**